topbar.js

```diff
@@ -201,11 +201,12 @@ class ColorEntry {
  */
 var FocusIcon = Utils.registerClass(
 class FocusIcon extends St.Icon {
-    _init(properties = {}, tooltip_x_point=0) {
+    _init(properties = {}, tooltip_parent, tooltip_x_point=0) {
         super._init(properties);
         this.reactive = true;
 
         // allow custom x position for tooltip
+        this.tooltip_parent = tooltip_parent ?? this;
```

Comment:
I like using `??` over `||`, but it would be good to check which version of gnome-shell adds support for it. No sense in supporting compatibility for older versions of gnome shell (e.g. looking for swipeTracker in different locations) if we're going to use js syntax that's not supported on those versions.

Reply:
According to my very quick research (take with a grain of salt): ?? support was added in GJS v1.65.90 (https://gitlab.gnome.org/GNOME/gjs/-/blob/master/NEWS#L993), Gnome version 40.4 was still using GJS 1.65.1 (https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/40.4/meson.build#L26) and Gnome version 41.1 is using GJS version 1.69.2 (https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/41.1/meson.build#L27).

"Officially" (according to the readme) we only support gnome 43 (and plan to support 44) on develop. However I'm personally also using gnome 42 on my work laptop. I think we should maybe decide or state more clearly which versions are actively supported.

Reply:
Good find @Lythenas!

So it looks like suppport for `??` and `?.` were the added/applied in gnome `41.0`:

https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/40.10/meson.build#L26
vs.
https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/41.0/meson.build#L27

Reply:
So, technically, those changes should be fine in `41` onwards but we haven't tested it.  Sounds like @Lythenas has tested/used the latest stuff in `42`.

I'm torn here, it can be quite a bit of work to support older versions (especially if we're not running those older version), and in some cases can limit new implementations and improvements (that's prob why I like rolling-releases in general... since I maintain some packages in the AUR as well).

Reply:
Let's carry on this thread in zulip (I was going to put up a thread approach for when new gnome versions come out... e.g. for `44` do we just then take a snapshot of the current dev in a branch called `gome-43` and that's what people get if they're on `43`?).